kod/object/active/holder/room.kod

```diff
@@ -2419,6 +2421,35 @@ messages:
       }
 
       propagate;
+   }  
+   
+   UserEntered(who = $)
+   {
+      local i,each_obj,lastWelcomeTime;
+
+      if isClass(who,&Player)
+         AND (send(who,@CheckPlayerFlag,#flag=PFLAG_ANONYMOUS)
+              OR send(who,@CheckPlayerFlag,#flag=PFLAG_INVISIBLE))
+      {
+         % q: is this check enough top skip players/admins
+         debug("Skipping Invisible / anonymous player");
+         return;
+      }
+
+      for i in plActive
+      {
+         each_obj = First(i);
+         % q: is there a better class than %Monster for all NPCs
+         if IsClass(each_obj, &Monster)
+         {
+            lastWelcomeTime = Send(each_obj, @GetLastWelcomeTime);
+            if GetTime() > (lastWelcomeTime + 30)
```

Comment:
I think this check belongs in Monster, not Room, since it's a behavior of the monster.